File: questik/app/src/main/java/ru/rinet/questik/data/room/AppDatabase.java
package ru.rinet.questik.data.room;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import ru.rinet.questik.data.room.model.Category;
import ru.rinet.questik.data.room.model.CategoryDao;
import ru.rinet.questik.data.room.model.Job;
import ru.rinet.questik.data.room.model.JobDao;
import ru.rinet.questik.data.room.model.Metrics;
import ru.rinet.questik.data.room.model.MetricsDao;
import ru.rinet.questik.data.room.model.Project;
import ru.rinet.questik.data.room.model.ProjectDao;
import ru.rinet.questik.data.room.model.Users;
import ru.rinet.questik.data.room.model.UsersDao;
import ru.rinet.questik.utils.LocalDateTypeConverter;

@Database(entities = {Job.class, Category.class, Metrics.class, Project.class, Users.class}, version = 1)
@TypeConverters({LocalDateTypeConverter.class})
public abstract class AppDatabase extends RoomDatabase {
    public abstract JobDao jobDao();
    public abstract CategoryDao categoryDao();
    public abstract MetricsDao metricsDao();
    public abstract ProjectDao projectDao();
    public abstract UsersDao usersDao();
}

File: app/src/main/java/ru/rinet/questik/repo/local/room/AppDatabase.kt
package ru.rinet.questik.repo.local.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import ru.rinet.questik.repo.local.room.dao.*
import ru.rinet.questik.repo.local.room.entity.*

@Database(entities = [CategoryEntity::class, JobsEntity::class, MaterialEntity::class, MetricsEntity::class, UserEntity::class,RemoteKeys::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun categoryDao(): CategoryDao
    abstract fun jobsDao(): JobsDao
    abstract fun materialDao(): MaterialDao
    abstract fun metricsDao(): MetricsDao
    abstract fun userDao(): UserDao
    abstract fun remoteKeysDao(): RemoteKeysDao

    companion object{
        @Volatile private var instance:AppDatabase? = null
        fun getDatabase(context: Context):AppDatabase =
            instance?: synchronized(this){
                instance?: buildDatabase(context).apply { instance = this }
            }

        private fun buildDatabase(context: Context) =
            Room.databaseBuilder(context,AppDatabase::class.java,"questik_database.db")
                .fallbackToDestructiveMigration()
                .build()
    }
}